### PR TITLE
Add windows/linux keybinding format

### PR DESCRIPTION
**Pull request by the-jasoney**
Release Notes:

- Added Windows and Linux keybinding format (e.g. Ctrl+Alt+Delete)

Screenshot 2024-03-09 212429

### CODE REVIEW

**Comment by the-jasoney** [comment]
Should the arrow keys be
1. Up, Down, Left, Right
2. Up Arrow, Down Arrow, Left Arrow, Right Arrow or
3. Arrow Up, Arrow Down, Arrow Left, Arrow Right

**Comment by the-jasoney** [comment]
What I'm concerned with right now is that the keybinding and the rest of the text can be too close together (like the Copy Relative Path item above)

**Comment by the-jasoney** [comment]
I adjusted the margins and it looks like this now:
Screenshot 2024-03-10 145057
compared to
Screenshot 2024-03-09 212429

**Comment by phisch** [comment]
@the-jasoney While you are at it, it looks like either the shortcut is a pixel to high up, or the label on the left is a pixel too low:
image

Maybe that's a little detail you could fix with this?

**Comment by bbb651** [comment]
I think we should change to the color to be slightly gray like a disabled item, firefox and gtk seem to do something similar:

Firefox menubar

Nautilus hamburger menu

QT on the other hand seems to just make the width of the dropdown big enough that there will be sufficient space between them:

Qalculate hamburger menu

**Comment by the-jasoney** [comment]
> I think we should change to the color to be slightly gray like a disabled item, firefox and gtk seem to do something similar:
> 
> Firefox menubar
> 
> Nautilus hamburger menu
> 
> QT on the other hand seems to just make the width of the dropdown big enough that there will be sufficient space between them:
> 
> Qalculate hamburger menu

Currently it uses the `text_muted` color, which with this theme makes it look slightly grayer than most text. Maybe we should increase the margin like Qt or use something like the `text_disabled` color.

**Comment by the-jasoney** [comment]
> @the-jasoney While you are at it, it looks like either the shortcut is a pixel to high up, or the label on the left is a pixel too low: image
> 
> Maybe that's a little detail you could fix with this?

After moving the shortcut down a pixel, I noticed that using some fonts like Zed Mono/Code and Segoe UI looked okay, but other fonts like Inter looked to be one pixel down. 

After applying commit a878d8f, with Zed Mono (Segoe UI was also like this):
Screenshot 2024-03-10 220317

After applying commit a878d8f, with Inter (Fira Code was also like this):
Screenshot 2024-03-10 220533

File: crates/ui/src/components/keybinding.rs [hunk 1]
```diff
@@ -16,31 +16,94 @@ impl RenderOnce for KeyBinding {
...
-                        el.child(KeyIcon::new(IconName::Command))
-                    })
-                    .when(keystroke.modifiers.shift, |el| {
-                        el.child(KeyIcon::new(IconName::Shift))
-                    })
-                    .when_some(key_icon, |el, icon| el.child(KeyIcon::new(icon)))
-                    .when(key_icon.is_none(), |el| {
-                        el.child(Key::new(keystroke.key.to_uppercase().clone()))
-                    })
+                if cfg!(target_os = "macos") {
```
Comment (maxdeviant) [inline crates/ui/src/components/keybinding.rs:19]: I'm not a huge fan of doing conditional compilation like this within a UI component.

I think we need a different way of achieving this.

Comment (jansol) [reply crates/ui/src/components/keybinding.rs:19]: Something like a get_key_hint(key_name) -> enum { KeyIcon, Text }?

**Comment by jansol** [comment]
> Currently it uses the `text_muted` color, which with this theme makes it look slightly grayer than most text. Maybe we should increase the margin like Qt or use something like the `text_disabled` color.

I'd argue that `text_muted` is indeed the right choice, semantically speaking. How much themes agree on that is another question.

File: crates/ui/src/components/keybinding.rs (hunk 1 above)
Comment (the-jasoney) [reply crates/ui/src/components/keybinding.rs]: (messily) resolved by bff9321

Comment (the-jasoney) [reply crates/ui/src/components/keybinding.rs]: If anyone has suggestions for how to clean up this code/move the conditional compilation out of the component, I would appreciate it

Comment (mikayla-maki) [reply crates/ui/src/components/keybinding.rs]: We'll be looking into how to do this in a more comprehensive way soon, the big concern is how do we test specialized windows / linux UI from our predominantly macOS dev team? That's a bit of a bigger question than this PR can handle on it's own. But once we've got a good approach, I'll let y'all know :)

**Comment by maxdeviant** [comment]
I opened #9192 for adding the general structure of platform-specific keybindings.

**Comment by the-jasoney** [comment]
Closed in favor of #9192

**Pull request closed without merging**